MapboxCoreNavigation/RouteController.swift

```diff
@@ -405,6 +197,20 @@ open class RouteController: NSObject {
         locationManager.stopUpdatingHeading()
         locationManager.delegate = nil
     }
+    
+    /**
+     Ends the current navigation session.
+     */
+    @objc public func endNavigation(feedback: EndOfRouteFeedback? = nil) {
+        if !didSendCancelEvent {
```

Comment:
AFAIK, `RouteController` is not designed to be reused at this time.

Reply:
@akitchen hmm, so yeah in talking with @bsudekum , it sounds like it might be possible (although out of design scope) to reuse the `RouteController`, possibly by changing `RouteController.routeProgress` (as you’re replacing the thing that the Routecontroller is iterating over)… I could reset that flag on `routeProgress`’ `didSet` as a defensive measure, how's that sound?